### ISSUE TITLE
multi_match accepts an array of values for "query"

### ISSUE THREAD

**Issue by GlenRSmith** [elastic/elasticsearch#20785]
**Elasticsearch version**: 2.3

A multi_match query that is otherwise proper, but has an array of values for the "query" parameter will execute successfully, but only actually perform the search using the last value of the array.

Should it return an error instead?

**Steps to reproduce**:

```
# cleanup
DELETE and_match
# create index/mapping
POST and_match
{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 0
  },
  "mappings": {
    "type1": {
      "properties": {
        "tags": {
          "type": "string"
        }
      }
    }
  }
}
# index
POST and_match/_bulk
{ "index" : { "_type" : "type1"} }
{ "tags" : "machine"}
{ "index" : { "_type" : "type1"} }
{ "tags" : "broken"}
{ "index" : { "_type" : "type1"} }
{ "tags" : "broken machine"}
# search
POST /and_match/_search
{
    "query": {
        "multi_match": {
          "query": ["broken", "machine"], 
          "fields": ["tags"]
        }
    }
}
```

Search results don't include the 2nd of the 3 documents.

**Comment by rjernst** [comment]
@GlenRSmith Can you try with the 5.0 beta? From glancing at the code (which was heavily refactored for 5.0) I believe it will now produce an error.

**Comment by GlenRSmith** [comment]
@rjernst Yep. In 5.0.0 beta1, the search produces an error.

**Comment by cbuescher** [comment]
Since this is already fixed in 5.0 I opened against 2.4 only. The fix is quiet straight forward, it should throw a QueryParsingException when encountering an array after any parameter other than the "fields" parameter.

**Comment by cbuescher** [comment]
Closed by 91f94e6 on 2.4 branch.
